extensions/binding/org.eclipse.smarthome.binding.onewire.test/src/test/java/org/eclipse/smarthome/binding/onewire/device/DS1923Test.java

```diff
@@ -0,0 +1,145 @@
+
+    @Before
+    public void setupMocks() {
+        setupMocks(THING_TYPE_MS_TH);
+        deviceTestClazz = DS1923.class;
+
+        addChannel(CHANNEL_TEMPERATURE, "Number:Temperature");
+        addChannel(CHANNEL_HUMIDITY, "Number:Dimensionless");
+        addChannel(CHANNEL_ABSOLUTE_HUMIDITY, "Number:Density");
+        addChannel(CHANNEL_DEWPOINT, "Number:Temperature");
+        addChannel(CHANNEL_SUPPLYVOLTAGE, "Number:Voltage");
+    }
+
+    @Test
+    public void presenceTestOn() {
+        presenceTest(OnOffType.ON);
+    }
+
+    @Test
+    public void presenceTestOff() {
+        presenceTest(OnOffType.OFF);
+    }
+
+    @Test
+    public void temperatureChannel() {
+        instantiateDevice();
+
+        try {
+            Mockito.when(mockBridgeHandler.checkPresence(testSensorId)).thenReturn(OnOffType.ON);
+            Mockito.when(mockBridgeHandler.readDecimalType(eq(testSensorId), any())).thenReturn(new DecimalType(10.0));
+
+            testDevice.enableChannel(CHANNEL_TEMPERATURE);
+            testDevice.configureChannels();
+            inOrder.verify(mockThingHandler).getThing();
+            testDevice.refresh(mockBridgeHandler, true);
+
+            inOrder.verify(mockBridgeHandler).readDecimalType(eq(testSensorId), any());
+            inOrder.verify(mockThingHandler).postUpdate(eq(CHANNEL_TEMPERATURE), eq(new QuantityType<>("10.0 °C")));
+
+            inOrder.verifyNoMoreInteractions();
+        } catch (OwException e) {
+            Assert.fail("caught unexpected OwException");
+        }
+    }
+
+    @Test
+    public void humidityChannel() {
+        instantiateDevice();
+
+        try {
+            Mockito.when(mockBridgeHandler.checkPresence(testSensorId)).thenReturn(OnOffType.ON);
+            Mockito.when(mockBridgeHandler.readDecimalType(eq(testSensorId), any())).thenReturn(new DecimalType(10.0));
+
+            testDevice.enableChannel(CHANNEL_HUMIDITY);
+            testDevice.enableChannel(CHANNEL_ABSOLUTE_HUMIDITY);
+            testDevice.enableChannel(CHANNEL_DEWPOINT);
+            testDevice.configureChannels();
+            inOrder.verify(mockThingHandler).getThing();
+            testDevice.refresh(mockBridgeHandler, true);
+
+            inOrder.verify(mockBridgeHandler, times(2)).readDecimalType(eq(testSensorId), any());
+            inOrder.verify(mockThingHandler).postUpdate(eq(CHANNEL_HUMIDITY), eq(new QuantityType<>("10.0 %")));
+            inOrder.verify(mockThingHandler).postUpdate(eq(CHANNEL_ABSOLUTE_HUMIDITY),
+                    eq(new QuantityType<>("0.9381970824113001000 g/m³")));
+            inOrder.verify(mockThingHandler).postUpdate(eq(CHANNEL_DEWPOINT),
+                    eq(new QuantityType<>("-20.31395053870025 °C")));
+
+            inOrder.verifyNoMoreInteractions();
+        } catch (OwException e) {
+            Assert.fail("caught unexpected OwException");
+        }
+    }
+
+    @Test
+    public void supplyVoltageChannel() {

```

Comment:
This test shows that there is no `supplyvoltage` channel. Is there a way how to verify that differently?

Reply:
I was thinking of a test that shows that the `supplyvoltagechannel` is removed from the thing by the `MultisensorThingHandler`. But that may be difficult because of the deferred channel configuration. I'll add something along these lines in my refactoring of the Multisensor. You can remove this test here.

Reply:
Test is removed, however I think that `ms-th` should not contain `supplyvoltagechannel` by default. It should be added dynamically to devices which really support it. It would simplify the logic.
I hope you will consider this during refactoring of Multisensor.